subsys/bluetooth/host/conn.c

```diff
@@ -1949,6 +1951,149 @@ static void bt_conn_set_param_le(struct bt_conn *conn,
 	conn->le.timeout = param->timeout;
 }
 
+#if defined(CONFIG_BT_WHITELIST)
+int bt_le_whitelist_add(const bt_addr_le_t *addr)
```

Comment:
Maybe these functions could do:
```
if (!IS_ENABLED(CONFIG_BT_WHITELIST)) {
   return -ENOTSUP;
}
```
instead of using preprocessor. As this is not enabled by default it would be safer to let it compiler (and check code) every time.

Reply:
@pdunaj Since the variables in `bt_dev` are compiled out to save memory. This code cannot be compiled without using preprocessor.
The test application for bluetooth shell enables the BT_WHITELIST, so it will be compiled as part of CI.

Reply:
ok